src/HTSPDemuxer.cpp

```diff
@@ -309,13 +309,8 @@ bool CHTSPDemuxer::IsRealTimeStream() const
    * we want the calculation to be consistent */
   CLockObject lock(m_mutex);
 
-  if (m_timeshiftStatus.shift == 0)
-    return true;
-
-  if (m_timeshiftStatus.start - m_timeshiftStatus.shift < 10)
-    return true;
-
-  return false;
+  /* Handle as real time when reading close to the EOF (10000000�s - 10s) */
+  return (m_timeshiftStatus.shift < 10000000);
```

Comment:
demux buffers are 8 secs so if you have less than this in your timeshift buffer, kodi's demux buffers can't fill. I don't know what the variables here mean and what the units are. vnsi considers a stream realtime if less that 10 secs are in the buffer.